spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1384,14 +1384,29 @@ def find_connection_file(self, connection_file):
 
     # ---- Public API
     # -------------------------------------------------------------------------
+    @property
+    def infowidget(self):
+        """
+        This is necessary to prevent an error when, in some situations, Python
+        garbage collects _infowidget.
+
+        Notes
+        -----
+        * See spyder-ide/spyder#21509 and spyder-ide/spyder#23529
+        """
+        try:
+            return self._infowidget
+        except RuntimeError:
+            self.enable_infowidget = False
+            return None
```

Comment:
```suggestion
            self._infowidget = self._create_info_widget()
            return self._infowidget
        except AttributeError:
            return None
```

* I checked and we can recreate `_infowidget` in case there is a RuntimeError.
* The AttributeError is necessary to prevent an error when calling this method in case `_infowidget` was not created.